yt_dlp/extractor/zingmp3.py

```diff
@@ -51,45 +80,101 @@ def _extract_item(self, item, fatal):
             self.raise_no_formats(msg, expected=True)
         self._sort_formats(formats)
 
-        subtitles = None
         lyric = item.get('lyric')
         if lyric:
             subtitles = {
                 'origin': [{
                     'url': lyric,
                 }],
             }
+        else:
+            api = self.get_api_with_signature(name_api=SLUG_API.get("lyric"), param={
+                "ctime": TIMESTAMP,
+                "id": self.song_id})
+            info_lyric = self.download_json(api)
+            subtitles = {
+                'origin': [{
+                    'url': try_get(info_lyric, lambda x: x['data']['file']),
+                }],
+            }
 
         album = item.get('album') or {}
 
         return {
             'id': item_id,
             'title': title,
             'formats': formats,
-            'thumbnail': item.get('thumbnail'),
+            'thumbnail': item.get('thumbnail') or item.get('thumbnailM'),
             'subtitles': subtitles,
             'duration': int_or_none(item.get('duration')),
             'track': title,
-            'artist': item.get('artists_names'),
+            'artist': item.get('artistsNames') or item.get('artists_names'),
             'album': album.get('name') or album.get('title'),
-            'album_artist': album.get('artists_names'),
+            'album_artist': album.get('artistsNames') or album.get('artists_names'),
         }
 
     def _real_extract(self, url):
-        page_id = self._match_id(url)
-        webpage = self._download_webpage(
-            url.replace('://zingmp3.vn/', '://mp3.zing.vn/'),
-            page_id, query={'play_song': 1})
-        data_path = self._search_regex(
-            r'data-xml="([^"]+)', webpage, 'data path')
-        return self._process_data(self._download_json(
-            'https://mp3.zing.vn/xhr' + data_path, page_id)['data'])
+        mobj = re.match(self._VALID_URL, url)
+        self.song_id = mobj.group('id')
+        self.type_url = mobj.group('type')
```

Comment:
You can/must do
```suggestion
        song_id, type_url = self._match_valid_url(url).group('id', 'type')
```
Don't assign non-constant values to `self`!

Reply:
@Lesmiscore Done. Thank you.